### ISSUE TITLE
[REVIEW]: Econometrics.jl

### ISSUE THREAD

**Issue by whedon** [JuliaCon/proceedings-review#38]
**Submitting author:** @Nosferican (<a href="http://orcid.org/0000-0002-8406-6175">José Bayoán Santiago Calderón</a>)
**Repository:** <a href="https://github.com/Nosferican/Econometrics.jl" target ="_blank">https://github.com/Nosferican/Econometrics.jl</a>
**Editor:** @matbesancon
**Reviewer:** @cpfiffer, @pkofod
**Archive:** <a href="https://doi.org/10.5281/zenodo.3380041" target="_blank">10.5281/zenodo.3380041</a>

## Status

[![status](https://proceedings.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb/status.svg)](https://proceedings.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb)

Status badge code:

```
HTML: <a href="https://submissions.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb"><img src="https://submissions.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb/status.svg"></a>
Markdown: [![status](https://submissions.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb/status.svg)](https://submissions.juliacon.org/papers/446fde271579d85e0d4c691d54093dbb)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Nosferican/Econometrics.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@cpfiffer & @pkofod, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/JuliaCon/proceedings-reviews/invitations

The reviewer guidelines are available here: https://proceedings.juliacon.org/guide/reviewers. Any questions/concerns please let @matbesancon know.

✨ **Please try and complete your review in the next two weeks** ✨

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Nosferican/Econometrics.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@Nosferican) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JuliaCon conflict of interest policy](https://proceedings.juliacon.org/guide/authors#coi) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JuliaCon code of conduct](https://proceedings.juliacon.org/coc).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Nosferican/Econometrics.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Authorship:** Has the submitting author (@Nosferican) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Paper format

- [x] **Authors:** Does the `paper.tex` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Page limit:** Is the page limit for full papers respected by the submitted document?

### Content

- [x] **Context:** is the scientific context motivating the work correctly presented?
- [x] **Methodology:** is the approach taken in the work justified, presented with enough details and reference to reproduce it?
- [x] **Results:** are the results presented and compared to approaches with similar goals?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @cpfiffer, @pkofod it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/JuliaCon/proceedings-review) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/JuliaCon/proceedings-review:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00038/jcon.00038/10.21105.jcon.00038.pdf)

**Comment by cpfiffer** [comment]
[econometrics-jl-comments.pdf](https://github.com/JuliaCon/proceedings-review/files/3419300/econometrics-jl-comments.pdf)

I've uploaded a PDF with some of my comments on the proof. Mostly minor stuff.

The remaining points I have on the review list are 

- API documentation. Your docstrings are pretty light in the actual source, and you don't have Documenter.jl set up to publish the docstrings on the site. I'd put a little time in to that if you have any.
- Community guidelines. The review checklist asks for this and it's not in the repo, so I'm not going to check it off. I'm also not super sure how important it is.

Generally good work. I had a fairly pleasant time working with it -- most everything seems quick and easy to use, and it's easy to shift between standard error types.

**Comment by Nosferican** [comment]
Thanks for taking time to go over it. I took a brief look at the PDF with the comments and will be making revisions in the next couple days.

I have updated the repository to docstring every functioned and struct defined in the package. Most of the functions lack docstring if those are defined in other packages and their docstring are appropriate.

The public API includes: `EconometricModel`, `absorb`, `BetweenEstimator`, `RandomEffectsEstimator`, `OIM`, `HC0`, `HC1`, `HC2`, `HC3`, `HC4`. In addition to the public API, there are a few `StatsBase.RegressionModel` API methods with modified methods (e.g., `fit` and `vcov`). Let me know if any are missing from the documentation.

The `README.md` have the shields for the documentation ([![Documentation: stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://nosferican.github.io/Econometrics.jl/stable) and [![Documentation: dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://nosferican.github.io/Econometrics.jl/dev)). You might have encountered a 500 error (GitHub was having issues today). Do let me know if for some reason the hosted documentation through Documenter.jl and GitHub Pages (domain .io) is not accessible to you.

I have also added the community guidelines to the repo.

**Comment by cpfiffer** [comment]
For the documentation API I was referring to using Documenter's abiity to export docstrings, not whether or not you have a documentation site. Currently it contains some useful examples examples, but no API library. 

Using the [StatsModels](https://juliastats.github.io/StatsModels.jl/stable/api/) site as an example, they have an `api.md` that uses Documenter to render all their docstrings to the website. It's pretty straightforward to add it to your site -- here's the [StatsModels api.md file](https://github.com/JuliaStats/StatsModels.jl/blob/master/docs/src/api.md) and their [make file](https://github.com/JuliaStats/StatsModels.jl/blob/master/docs/make.jl).

Doing this would make it a lot easier for people to click around your documentation site and see how certain things work, what they need, what they do, etc. 

At this point, it's only really a suggestion. I've checked off the last box for the review. I think having the docstrings as they are right now is satisfactory, but I would recommend publishing the API on the site going forward.

Great work!

**Comment by Nosferican** [comment]
Got it. Thanks for the feedback. I think the public API is stable enough to make it "official" and publish it in the docs now (e.g., the internals of the types). Will do.

**Comment by pkofod** [comment]
I have a few comments for the paper and repo. Note that I did not check off all boxes as I do not believe everything is totally fulfilled.

**Installation**
There are two boxes that relate to installation: one for the docs and one for the paper. Neither provide instructions about how to actually install the software.

**Statement of need**
The paper does clearly state what the point of the software is, but the documentation actually doesn't.

**DOIs**
Not all papers, books, etc have DOIs. One example is https://www.springer.com/kr/book/9783540755708 that has no doi listed, but actually has one.

**Page limit**
Is there a page limit? I couldn't find one, so I'm not sure. @matbesancon 

**Paper**
Mostly good. I came across one paragraph that stood out as odd/unnecessary:
```
The Julia language [3] is an upcoming language especially well-
suited for scientific computing such as econometrics, data science,
machine learning, and other related tasks.
```
I suggest you delete it.

Could you briefly discuss/mention what the point of `EconometricModel` is? When the reader gets to the figures, these come a bit out of the blue; I tried to go back and see if I had missed their introduction! It's "shown" how to use the software, but not really explained. I'm not requesting 2 pages, but just something that says what goes where, etc. It's not clear to the uninitiated - though I obviously don't expect a full "documentation" repeated in the paper.

**Comment by Nosferican** [comment]
Thanks for the review.

**Paper Limit**
The normal paper is 6-8 pages according to https://github.com/JuliaCon/proceedings.juliacon.org/pull/9#issuecomment-511241172.

**On the issue of DOI for references.**
The DOI are included in the source (see [ref.bib](https://github.com/Nosferican/Econometrics.jl/blob/master/paper/ref.bib#L393)), but the JuliaCon template does not display those (see https://github.com/JuliaCon/JuliaConSubmission.jl/issues/7).

I will be working on incorporating the feedback I received from y'all next week. I will ping y'all and share the revisions. Thanks!

**Comment by Nosferican** [comment]
@cpfiffer & @pkofod I have updated the paper and docs to address the feedback. Thanks!

**Comment by pkofod** [comment]
@matbesancon can whedon re-compile the pdf ?

**Comment by matbesancon** [comment]
@whedon build pdf

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00038/jcon.00038/10.21105.jcon.00038.pdf)

**Comment by cpfiffer** [comment]
Cool, I'm happy.

**Comment by pkofod** [comment]
Can you perhaps tell me in what way you addressed the issue of installation instructions @Nosferican ? I can't see any changes to either the readme, docs or paper on installation. Thanks.

**Comment by Nosferican** [comment]
Absolutely.
- [docs](https://nosferican.github.io/Econometrics.jl/dev/getting_started/#Installation-1) (right now in the dev, but as soon as I tag a new release it will be updated to the stable as well)
- [paper](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00038/jcon.00038/10.21105.jcon.00038.pdf) I added to the conclusion how to obtain the software next to the license information.
> The package is registered in the official Julia registry (General); as such it can be installed using the Julia package manager.
- [readme](https://github.com/Nosferican/Econometrics.jl), I added a shield with the information of the latest release. I don't know if it should be more explicit that it is the release in the General registry. If you think it should have something more explicit about it being registered with General or the basic installation with the Pkg manager I would be happy to include it if deemed helpful (e.g., linking to the docs for the installation instructions). I have both Registrator.jl and TagBot.jl enabled to keep both in synch.

**Comment by pkofod** [comment]
I didnt look at the dev version of the docs, it looks fine.

**Comment by pkofod** [comment]
@matbesancon I think that's a :+1: from @cpfiffer and I then.

**Comment by matbesancon** [comment]
Cool thanks all, I'll look at everything tomorrow morning

**Comment by Nosferican** [comment]
Thanks again for all the good feedback!

**Comment by Nosferican** [comment]
Let me know if there is anything that needs addressing.

**Comment by matbesancon** [comment]
@whedon list commands

**Comment by matbesancon** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to accept the paper and deposit with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

```

**Comment by matbesancon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.3386/w24003 is OK
- 10.1111/j.1468-0084.1987.mp49004006.x is OK
- 10.2307/144855 is OK
- 10.3368/jhr.50.2.317 is OK
- 10.2307/2118326 is OK
- 10.1055/s-0030-1267099 is OK
- 10.3386/w20342 is OK
- 10.1093/biomet/73.1.13 is OK
- 10.2307/1912934 is OK
- 10.1016/0304-4076(85)90158-7 is OK
- 10.2307/2525981 is OK
- 10.3368/jhr.50.2.301 is OK
- 10.1001/jamasurg.2018.0035 is OK
- 10.1111/j.0012-9682.2008.00821.x is OK
- 10.1161/CIRCULATIONAHA.116.024247 is OK
- 10.18128/D030.V6.0 is OK
- 10.1073/pnas.1510489113 is OK
- 10.1257/jep.31.2.3 is OK
- 10.1177/0193841X04266335 is OK
- 10.1257/aer.96.4.988 is OK
- 10.1162/REST_a_00662 is OK
- 10.18128/D030.V6.0 is OK
- 10.1257/000282805774670130 is OK
- 10.3368/jhr.XL.4.969 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.1090/S0025-5718-1977-0428694-0 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.2592615 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK
- 10.2307/1392405 is OK

MISSING DOIs

- https://doi.org/10.1016/bs.hefe.2016.10.003 may be missing for title: The Econometrics of Randomized Experiments
- https://doi.org/10.5642/cguetd/132 may be missing for title: On Cluster Robust Models
- https://doi.org/10.1080/01621459.2017.1319839 may be missing for title: Estimation and Inference of Heterogeneous Treatment Effects using Random Forests
- https://doi.org/10.3386/w21875 may be missing for title: Multiple Hypothesis Testing in Experimental Economics

INVALID DOIs

- 10.2307/25800796 is INVALID
```

**Comment by matbesancon** [comment]
@Nosferican can you check the missing or invalid DOIs above?

**Comment by Nosferican** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00038/jcon.00038/10.21105.jcon.00038.pdf)

**Comment by Nosferican** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5642/cguetd/132 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.2592615 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by matbesancon** [comment]
wonderful, I'll have a final read tomorrow

**Comment by Nosferican** [comment]
Thanks. Also @whedon, you such a good bot.

**Comment by matbesancon** [comment]
@Nosferican some remarks:
Page 2:
footnote 1: there is no A and b here, so it may be hard to understand what you mean. Something like: " `\` is a linear solver operator, such that `x = A\b => Ax = b`
Equation 3: I assume `.*` is a broadcasted multiplication operator? It should be specified.

Page 3:
"solving these problemS" (typo)
Equation 11: maybe without using a double letter for a single variable? It can be confusing
Section 2.5 title: First-Difference Estimator (not just First-Difference)

Reference [2]: use double bracket `{{}}` to maintain the capitalized MixedModels.jl

**Comment by matbesancon** [comment]
Best practices: you can add a footnote link to the documentation

**Comment by matbesancon** [comment]
Page 4:
> The Julia ecosystem follows this canonical pipeline.

The sentence is confusing, the Julia statistics and data ecosystems enable / provide features for the whole pipeline?

**Comment by matbesancon** [comment]
Don't forget to properly cite the packages you mention.
This is more of a personal opinion, so you may take it with a grain of salt, but I find the "Design Decisions" section to be a bit informal. I would rephrase the "Should software be dummy-proof?" question for instance

**Comment by Nosferican** [comment]
Thanks for the feedback. I think I have gone over all the issues. For packages that do not provide any suggested citation such as no Zedeno/article (e.g., StatsModels.jl / DataFrames.jl), any suggestion on how to cite them?

**Comment by matbesancon** [comment]
What I usually do is something like:
```
@misc{kerneldensity,
    author = {{JuliaStats}},
    title = {{KernelDensity.jl}},
    url = {https://github.com/JuliaStats/KernelDensity.jl},
    version = {0.4.1},
    year = {2018},
    month= {02}
}
```

Feel free to replace the authors with the maintainers' names

**Comment by Nosferican** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/JuliaCon/proceedings-papers/blob/jcon.00038/jcon.00038/10.21105.jcon.00038.pdf)

**Comment by Nosferican** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5642/cguetd/132 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.2592615 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK
- 10.5281/zenodo.3376013 is OK
- 10.5281/zenodo.3376171 is OK
- 10.5281/zenodo.3376169 is OK
- 10.5281/zenodo.3376179 is OK
- 10.5281/zenodo.3376177 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by matbesancon** [comment]
@Nosferican re-check the comments above, some were not addressed (solving these problemS), and others

**Comment by matbesancon** [comment]
Also, I think you haven't put the package citations

**Comment by matbesancon** [comment]
there is a weird effect of two columns almost merging in the middle. @Nosferican do you get the same effect locally?

**Comment by matbesancon** [comment]
Wonderful that seems fixed

**Comment by matbesancon** [comment]
ok @Nosferican this looks good to me, you can do a last check, the last things:
I would advise releasing a new version, then putting it on Zenodo to get a DOI, the DOI will be attached to the paper

Once this is done, I will set the version, DOI and figure out how to publish

**Comment by matbesancon** [comment]
@whedon set v0.2.2 as version

**Comment by matbesancon** [comment]
@whedon set 10.5281/zenodo.3380041 as archive

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by matbesancon** [comment]
@whedon accept

**Comment by matbesancon** [comment]
that's bizarre. Are you using this `csquotes.sty`?

**Comment by matbesancon** [comment]
could you add it to the dependencies, it might not be included in the server's build

**Comment by matbesancon** [comment]
In the `paper` folder, simply add the corresponding `sty` file which should be easy to find online

**Comment by matbesancon** [comment]
@whedon generate pdf

**Comment by matbesancon** [comment]
@whedon accept

**Comment by matbesancon** [comment]
I think csquotes.def and *.cfg are also required, sorry!

**Comment by matbesancon** [comment]
@whedon accept

**Comment by matbesancon** [comment]
thank you yes

**Comment by matbesancon** [comment]
@whedon accept

**Comment by matbesancon** [comment]
@vchuravy have you had this in other submissions?

**Comment by Nosferican** [comment]
Should I try not using `csquotes`?

**Comment by matbesancon** [comment]
Yes please, I sadly don't have a good explanation here

**Comment by Nosferican** [comment]
I updated the repo... Maybe try it in half an hour?

**Comment by matbesancon** [comment]
I'll recompile yes thanks

**Comment by Nosferican** [comment]
Should be good now to see if the `accept` complains or not.

**Comment by matbesancon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/4

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/4, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by matbesancon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @matbesancon, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by matbesancon** [comment]
@vchuravy woops

**Comment by Nosferican** [comment]
Bump.

**Comment by matbesancon** [comment]
@Nosferican all good here, congratulations, we'll re-bundle and build everything when we have issue one

**Comment by Nosferican** [comment]
Any update on the issue 1 timeline?

**Comment by matbesancon** [comment]
hi, we are figuring out the last steps with pushing out the issue, sorry for the delays (pinging @vchuravy if more information is available)

**Comment by Nosferican** [comment]
Any updates on the timeline?

**Comment by matbesancon** [comment]
sorry about the delay, @vchuravy is working on the website hosting the articles, which is in the last steps to finally have thing working

**Comment by vchuravy** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.5642/cguetd/132 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.596435 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK
- 10.5281/zenodo.3376013 is OK
- 10.5281/zenodo.3376171 is OK
- 10.5281/zenodo.3376169 is OK
- 10.5281/zenodo.3376179 is OK
- 10.5281/zenodo.3376177 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Nosferican** [comment]
Any word on when issue 1 will be out?

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #38 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon.rb:200:in `block in parse_authors': Author (José Bayoán Santiago Calderón) is missing affiliation (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon.rb:198:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon.rb:198:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon.rb:91:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:81:in `compile'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.5642/cguetd/132 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.596435 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK
- 10.5281/zenodo.3376013 is OK
- 10.5281/zenodo.3376171 is OK
- 10.5281/zenodo.3376169 is OK
- 10.5281/zenodo.3376179 is OK
- 10.5281/zenodo.3376177 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@Nosferican - could you please merge this PR? https://github.com/Nosferican/Econometrics.jl/pull/49

**Comment by matbesancon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @JuliaCon/jcon-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/JuliaCon/proceedings-papers/pull/13

If the paper PDF and Crossref deposit XML look good in https://github.com/JuliaCon/proceedings-papers/pull/13, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.5642/cguetd/132 is OK
- 10.1137/141000671 is OK
- 10.1007/978-3-540-75571-5 is OK
- 10.1137/0611032 is OK
- 10.2307/1907330 is OK
- 10.3200/JECE.39.1.88-91 is OK
- 10.1137/10079687X is OK
- 10.2307/1909405 is OK
- 10.1017/S0266466600010318 is OK
- 10.18637/jss.v027.i02 is OK
- 10.18637/jss.v011.i10 is OK
- 10.5281/zenodo.596435 is OK
- 10.1080/0022250X.1975.9989847 is OK
- 10.21105/joss.00615 is OK
- 10.2307/2109893 is OK
- 10.5281/zenodo.3376013 is OK
- 10.5281/zenodo.3376171 is OK
- 10.5281/zenodo.3376169 is OK
- 10.5281/zenodo.3376179 is OK
- 10.5281/zenodo.3376177 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by matbesancon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JCON!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/JuliaCon/proceedings-papers/pull/14
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/jcon.00038](https://doi.org/10.21105/jcon.00038)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00038/status.svg)](https://doi.org/10.21105/jcon.00038)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jcon.00038">
  <img src="https://proceedings.juliacon.org/papers/10.21105/jcon.00038/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://proceedings.juliacon.org/papers/10.21105/jcon.00038/status.svg
   :target: https://doi.org/10.21105/jcon.00038
```
This is how it will look in your documentation:

[![DOI](https://proceedings.juliacon.org/papers/10.21105/jcon.00038/status.svg)](https://doi.org/10.21105/jcon.00038)

**We need your help!**

JuliaCon Proceedings is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://forms.gle/yANMe4c9YmuW9MWcA
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-jcon
